uberenv.py

```diff
@@ -455,30 +469,45 @@ def clean_build(self):
                         res = sexe(unist_cmd, echo=True)
 
     def show_info(self):
-        spec_cmd = "spack/bin/spack spec " + self.pkg_name + self.opts["spec"]
-        return sexe(spec_cmd, echo=True)
+        spec_cmd = "spack/bin/spack spec -IL " + self.pkg_name + self.opts["spec"]
+
+        res, out = sexe(spec_cmd, ret_output=True, echo=True)
+        print(out)
+
+        for line in out.split("\n"):
+            if re.match(r"^(\[\+\]| - )  [a-z0-9]{32}  " + re.escape(self.pkg_name), line):
+                self.spec_hash = line.split("  ")[1]
+                if line.startswith("[+]"):
+                    pkg_path = self.find_spack_pkg_path_form_hash(self.pkg_name,self.spec_hash)
+                    install_path = pkg_path["path"]
+                    if os.path.isdir(install_path):
+                        print("[Warning: {} {} has already been install in {}]".format(self.pkg_name, self.opts["spec"],install_path))
+                        print("[Warning: Uberenv will proceed using this directory]".format(self.pkg_name))
+                        self.opts["use_install"] = True
+
+        return res
 
     def install(self):
         # use the uberenv package to trigger the right builds
         # and build an host-config.cmake file
-        install_cmd = "spack/bin/spack "
-        if self.opts["ignore_ssl_errors"]:
-            install_cmd += "-k "
-        if not self.opts["install"]:
-            install_cmd += "dev-build -d {} -u {} ".format(self.pkg_src_dir,self.pkg_final_phase)
-        else:
-            install_cmd += "install "
-            if self.opts["run_tests"]:
-                install_cmd += "--test=root "
-        install_cmd += self.pkg_name + self.opts["spec"]
-        res = sexe(install_cmd, echo=True)
-        if res != 0:
-            return res
 
+        if not "use_install" in self.opts:
```

Comment:
Is the `use_install` option documented?

Reply:
For some reason I don’t remember, I added this as an option but only used and assignable internally, so not really an option.
I’m making is a `SpackEnv` field.

Reply:
Done